### PR TITLE
JS: Sharpen missing CSRF middleware query

### PR DESCRIPTION
**Pull request by asgerf**
Addresses the FP reported in https://github.com/Semmle/ql/issues/2539.

Evaluation underway

### CODE REVIEW

**Comment by asgerf** [comment]
Added tests and sharpened a bit further to avoid the FPs in https://github.com/Semmle/ql/issues/2476.

**Comment by asgerf** [comment]
[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/csrf-only-session-cookie-access_1579827584382) looks ok. Some of the handlers referencing `req.user` seemed worth keeping so I've pushed a commit to preserve those results.

File: javascript/ql/src/Security/CWE-352/MissingCsrfMiddleware.ql [hunk 1]
```diff
@@ -63,11 +113,23 @@ from
...
+  // Require that the handler uses cookies and has cookie middleware.
+  //
+  // In practice, handlers that use cookies always have the cookie middleware or
+  // the handler wouldn't work. However, if we can't find the cookie middleware, it
+  // indicates that our middleware model is too incomplete, so in that case we
+  // don't trust it to detect the presence of CSRF middleware either.
+  getARouteUsingCookies().flowsToExpr(handler) and
   hasCookieMiddleware(handler, cookie) and
+
+  // Only flag the first cookie parser registered first.
```
Comment (erik-krogh) [inline javascript/ql/src/Security/CWE-352/MissingCsrfMiddleware.ql]:
```suggestion
  // Only flag the cookie parser registered first.
```

**Comment by asgerf** [comment]
@max-schaefer could you re-approve or review again?

**Pull request merged**